### PR TITLE
[ENHANCEMENT] Story Mode Menu Mouse Scrolling

### PR DESCRIPTION
**Pull request by JVNpixels**
# Adds the ability to scroll in the StoryMenuState.
This pull request adds the ability for scrolling in the story mode menu, and it would be a good quality of life feature.

### CODE REVIEW

**Comment by Raltyro** [comment]
wait in html5 it adds scrolls by 8 instead of just one?

**Comment by JVNpixels** [comment]
Thats what the HTML5 code says.

File: source/funkin/ui/story/StoryMenuState.hx [hunk 1]
```diff
@@ -336,6 +336,24 @@ class StoryMenuState extends MusicBeatState
...
+        {
+          changeLevel(-Math.round(FlxG.mouse.wheel / 8));
+        }
+        else if (FlxG.mouse.wheel > 0)
+        {
+          changeLevel(-Math.round(FlxG.mouse.wheel / 8));
+        }
+        #end
+
+        // HTML and NON HTML builds mouse fix.
```
Comment (gamerbross) [inline source/funkin/ui/story/StoryMenuState.hx:355]: Everything looks great!
```suggestion
```
But this comment is not needed

**Comment by JVNpixels** [comment]
Comment has been removed.

**Comment by JVNpixels** [comment]
Is this good enough to merge?

File: source/funkin/ui/story/StoryMenuState.hx [hunk 2]
```diff
@@ -336,6 +336,23 @@ class StoryMenuState extends MusicBeatState
...
+        if (FlxG.mouse.wheel < 0)
+        {
+          changeLevel(-Math.round(FlxG.mouse.wheel / 8));
+        }
+        else if (FlxG.mouse.wheel > 0)
+        {
+          changeLevel(-Math.round(FlxG.mouse.wheel / 8));
+        }
+        #end
+
```
Comment (gamerbross) [inline source/funkin/ui/story/StoryMenuState.hx:354]: Yeah, all good. just a picky change you can do is remove the extra newline

**Review by gamerbross — APPROVED** [review]

**Comment by JVNpixels** [comment]
All should be fixed now!

**Comment by EliteMasterEric** [comment]
image

gamerbross approved that's how you know it's good

**Comment by EliteMasterEric** [comment]
https://github.com/FunkinCrew/Funkin/assets/4635334/5fd2ff4a-ee17-465e-b903-9eb1539b204b

10/10 peak content right here

**Review by EliteMasterEric — APPROVED** [review]

**Comment by JVNpixels** [comment]
Thank you!

**Comment by JVNpixels** [comment]
It will be merged internally, correct?

**Comment by JVNpixels** [comment]
Just to clarify, have these been merged yet or will this be for later updates?
Milestone / Goal: 0.5 or next update.

**Comment by NotHyper-474** [comment]
Generally when they accept it they merge it internally (~~as in merging into their ultra duper secret mirror repo~~)

**Comment by koueu** [comment]
could you do this but with the main menu?

**Comment by koueu** [comment]
is there a way to add the scroll as a bind in control.hx so it works on all menus?

**Comment by koueu** [comment]
image
like is there a action.ui_down_alt or anything similar

**Comment by JVNpixels** [comment]
I'm not sure, but they don't have mouse support for everything yet. Only the FreePlay or Story Mode if this PR gets merged next update.

**Comment by AbnormalPoof** [comment]
> if this PR gets merged next update.

Generally, PRs with the `status: accepted` label are set to be merged in the next update (0.5.0, as dictated by the milestone). So yes, it should be in the next update.

**Comment by JVNpixels** [comment]
Yo it is in the game!

**Pull request merged**